lib/beaker-puppet/install_utils/puppet_utils.rb

```diff
@@ -90,41 +90,51 @@ def get_puppet_collection(agent_version = 'latest')
           collection
         end
 
-        # Determine the puppet collection that matches a given version of the puppet-agent
-        # package (you can find this version in the `aio_agent_version` fact).
+        # Determine the puppet collection that matches a given package version. The package
+        # must be one of
+        #   * :puppet_agent (you can get this version from the `aio_agent_version_fact`)
+        #   * :puppet
         #
-        # @param agent_version [String] a semver version number of the puppet-agent package, or the string 'latest'
+        #
+        # @param package [Symbol] the package name. must be one of :puppet_agent, :puppet
+        # @param version [String] a version number or the string 'latest'
         # @returns [String|nil] the name of the corresponding puppet collection, if any
-        def puppet_collection_for_puppet_agent_version(agent_version)
-          agent_version = agent_version.to_s
-          return 'puppet' if agent_version.strip == 'latest'
+        def puppet_collection_for(package, version)
+          valid_packages = [
+            :puppet_agent,
+            :puppet
+          ]
 
-          x, y, z = agent_version.to_s.split('.').map(&:to_i)
-          return nil if x.nil? || y.nil? || z.nil?
+          unless valid_packages.include?(package)
+            raise "package must be one of #{valid_packages.join(', ')}"
+          end
 
-          return 'pc1' if x == 1
+          case package
+          when :puppet_agent
+            agent_version = version.to_s
+            return 'puppet' if agent_version.strip == 'latest'
 
-          # A y version >= 99 indicates a pre-release version of the next x release
-          x += 1 if y >= 99
-          "puppet#{x}" if x > 4
-        end
+            x, y, z = agent_version.to_s.split('.').map(&:to_i)
+            return nil if x.nil? || y.nil? || z.nil?
 
-        # Determine the puppet collection that matches a given version of the puppet gem.
-        #
-        # @param version [String] a semver version number of the puppet gem, or the string 'latest'
-        # @returns [String|nil] the name of the corresponding puppet collection, if any
-        def puppet_collection_for_puppet_version(puppet_version)
-          puppet_version = puppet_version.to_s
-          return 'puppet' if puppet_version.strip == 'latest'
+            return 'pc1' if x == 1
+
+            # A y version >= 99 indicates a pre-release version of the next x release
+            x += 1 if y >= 99
+            "puppet#{x}" if x > 4
+          when :puppet
+            puppet_version = version.to_s
+            return 'puppet' if puppet_version.strip == 'latest'
 
-          x, y, z = puppet_version.to_s.split('.').map(&:to_i)
-          return nil if x.nil? || y.nil? || z.nil?
+            x, y, z = puppet_version.to_s.split('.').map(&:to_i)
+            return nil if x.nil? || y.nil? || z.nil?
```

Comment:
Hmm, I feel like we have enough information to return a valid result if the only thing we have is `x`, since it's only the major version that defines the release stream. Though, I suppose if `y` and `z` are nil that might indicate a malformed version string.... I'm not sure my hang up on this is worth any follow up work

Reply:
We'll still need `y` to handle weird cases like the `5.99` tag (I forgot why we did that in the first place -- maybe @caseywilliams can clarify ?).

It might be worth raising an exception for a malformed version string?

Reply:
I feel like raising an error isn't a bad idea if we can't figure out what's happening with the version string. Do we have existing utilities available to help parse version strings?

Reply:
There is [Beaker::Shared::Semvar](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/shared/semvar.rb#L12), but that doesn't raise any errors if the versions are malformed strings and it isn't specific to semver parsing.